### ISSUE TITLE
Automatically use non-compute shaders when compute shader compilation fails

### ISSUE THREAD

**Issue by qwertychouskie** [supertuxkart/stk-code#2860]
When a compute shader fails to compile, STK should automatically use the non-compute version.

**Comment by MTres19** [comment]
I'm not sure this would actually work. The shaders might compile, but [still be bugged](https://github.com/supertuxkart/stk-code/issues/1916). I don't think shader compilation was the problem in that ticket.

**Comment by qwertychouskie** [comment]
It obviously wouldn't fix all issues, but would help in some cases.

**Comment by auriamg** [comment]
which cases? have you got data, logs, etc. showing some specific computers/drivers where they fail to compile and why? We need to decide how to proceed based on actual data to evaluate the proper course of action, not the hope that it might help in some cases

**Comment by qwertychouskie** [comment]
E.G. https://github.com/supertuxkart/stk-code/issues/2595

**Comment by auriamg** [comment]
@qwertychouskie as far as I can tell from that log, in this case there was no compilation error reported. your suggestion would thus be useless in that case. Besides, that issue was already fixed by deveee.

**Comment by qwertychouskie** [comment]
Actually that issue is still open (though @SuicSoft switched to different drivers that are not affected).  In that case, STK crashes on compiling the first compute shader.

**Comment by qwertychouskie** [comment]
So STK would have to handle exceptions in shader compilation, and handle it appropriately (disable compute if the exception is in a compute shader, use the OpenGL 2 pipeline if it fails in any other shader that is not using a disable-able feature).

**Comment by deveee** [comment]
Last time that I saw failed compute shaders compilation was rather long time ago and it's already fixed. Compute shaders sometimes cause crashes for radeon drivers, and also cause artifacts (i.e. horizontal lines on intel), but none of them are easy to catch when running the game. So we can't really do much more than adding some graphical restrictions. 

Closing as invalid because the suggestion IMO doesn't have a sense.
